django/core/signing.py

```diff
@@ -144,9 +162,19 @@ def loads(s, key=None, salt='django.core.signing', serializer=JSONSerializer, ma
 
 class Signer:
 
-    def __init__(self, key=None, sep=':', salt=None):
-        # Use of native strings in all versions of Python
-        self.key = key or settings.SECRET_KEY
+    def __init__(self, key=None, sep=':', old_keys=None, salt=None):
+        if key is None:
+            if old_keys is not None:
+                raise ImproperlyConfigured('You must specify key when you specify old_keys.')
+
+            self.key = get_secret_key()
+            self.old_keys = [self.key] + get_old_keys()
```

Comment:
`all_keys` ?

Reply:
This has changed in the latest revision and is handled in another way now.